set/set_test.go

```diff
@@ -1311,6 +1314,60 @@ func TestBackup(t *testing.T) {
 
 			testBackupOK(remotePath)
 		})
+
+		Convey("and can back it up to iRODS as well", func() {
+			remoteDir := os.Getenv("IBACKUP_TEST_COLLECTION")
+			if remoteDir == "" {
+				SkipConvey("skipping iRODS backup test since IBACKUP_TEST_COLLECTION not set", func() {})
+
+				return
+			}
+
+			remotePath := filepath.Join(remoteDir, "db")
+
+			handler, err := put.GetBatonHandler()
+			So(err, ShouldBeNil)
+
+			db.EnableRemoteBackups(remotePath, handler)
+
+			err = db.Backup()
+			So(err, ShouldBeNil)
+
+			localPath := t.TempDir()
+			localDB := filepath.Join(localPath, "db")
+
+			_, err = exec.Command("iget", remotePath, localDB).CombinedOutput()
+			So(err, ShouldBeNil)
+
+			testBackupOK(localDB)
+
+			Convey("and child baton-do processes end after upload", func() {
+				pid := strconv.FormatInt(int64(os.Getpid()), 10)
+				subtasks, err := filepath.Glob(filepath.Join("/proc", pid, "task", "*", "children"))
```

Comment:
One of our (indirect?) dependencies offers methods for looking at processes and children. Consider using it if it makes this test easier to read

Reply:
I'll take a look.